clock/README.md

```diff
@@ -5,6 +5,362 @@ A Go time testing library for writing deterministic unit tests
+
+    count := 0
+    go mClock.AfterFunc(time.Hour, func(){
+        count++
+    })
+    call := trap.MustWait(ctx)
+    call.Release()
+    if call.Duration != time.Hour {
+        t.Fatal("wrong duration")
+    }
+
+    // Now that the async call to AfterFunc has occurred, we can advance the clock to trigger it
+    mClock.Advance(time.Hour).MustWait(ctx)
+    if count != 1 {
+        t.Fatal("wrong count")
+    }
+}
+```
+
+In this test, the trap serves 2 purposes. Firstly, it allows us to capture and assert the duration
+passed to the `AfterFunc` call. Secondly, it prevents a race between setting the timer and advancing
+it. Since these things happen on different goroutines, if `Advance()` completes before
+`AfterFunc()` is called, then the timer never pops in this test.
+
+Any untrapped calls immediately complete using the current time, and calling `Close()` on a trap
+causes the mock clock to stop trapping those calls.
+
+You may also `Advance()` the clock between trapping a call and releasing it. The call uses the
+current (mocked) time at the moment it is released.
+
+```go
+func TestTrap2(t *testing.T) {
+    ctx, cancel := context.WithTimeout(10*time.Second)
+    defer cancel()
+    mClock := quartz.NewMock(t)
+    trap := mClock.Trap().Since()
+    defer trap.Close() // stop trapping AfterFunc calls
+
+    var logs []string
+    done := make(chan struct{})
+    go func(clk quartz.Clock){
+        defer close(done)
+        start := clk.Now()
+        phase1()
+        p1end := clk.Now()
+        logs = append(fmt.Sprintf("Phase 1 took %s", p1end.Sub(start).String()))
+        phase2()
+        p2end := clk.Now()
+        logs = append(fmt.Sprintf("Phase 2 took %s", p2end.Sub(p1end).String()))
+    }(mClock)
+
+    // start
+    trap.MustWait(ctx).Release()
+    // phase 1
+    call := trap.MustWait(ctx)
+    mClock.Advance(3*time.Second).MustWait(ctx)
+    call.Release()
+    // phase 2
+    call = trap.MustWait(ctx)
+    mClock.Advance(5*time.Second).MustWait(ctx)
+    call.Release()
+
+    <-done
+    // Now logs contains []string{"Phase 1 took 3s", "Phase 2 took 5s"}
+}
+```
+
+### Tags
+
+When multiple goroutines in the code under test call into the Clock, you can use `tags` to
+distinguish them in your traps.
+
+```go
+trap := mClock.Trap.Now("foo") // traps any calls that contain "foo"
+defer trap.Close()
+
+foo := make(chan time.Time)
+go func(){
+  foo <- mClock.Now("foo", "bar")
+}()
+baz := make(chan time.Time)
+go func(){
+    baz <- mClock.Now("baz")
+}()
+call := trap.MustWait(ctx)
+mClock.Advance(time.Second).MustWait(ctx)
+call.Release()
+// call.Tags contains []string{"foo", "bar"}
+
+gotFoo := <-foo // 1s after start
+gotBaz := <-baz // ?? never trapped, so races with Advance()
+```
+
+Tags appear as an optional suffix on all `Clock` methods (type `...string`) and are ignored entirely
+by the realtime clock. They also appear on all methods on returned timers and tickers.
```

Comment:
Wording, for me realtime clock suggests hardware component, perhaps worth a rewording?